frontend/src/components/grantaccess/GrantDelegatedAdmin.vue

```diff
@@ -0,0 +1,222 @@
+            <form id="grantAdminForm" class="form-container">
+                <StepContainer title="User information">
+                    <UserDomainSelect
+                        :domain="formData.domain"
+                        @change="userDomainChange"
+                    />
+                    <UserNameInput
+                        :domain="formData.domain"
+                        :userId="formData.userId"
+                        @change="userIdChange"
+                        @setVerifyResult="setVerifyUserIdPassed"
+                    />
+                </StepContainer>
+
+                <StepContainer
+                    title="Add the role a delegated admin can assign"
+                    :divider="isAbstractRoleSelected()"
+                >
+                    <label htmlFor="roleId">Assign a role to the user</label>
+
+                    <Field
+                        id="roleId"
+                        name="roleId"
+                        aria-label="Role Select"
+                        v-slot="{ field, handleChange }"
+                        v-model="formData.roleId"
+                    >
+                        <Dropdown
+                            :options="delegatedRoleOptions"
+                            optionLabel="name"
+                            optionValue="id"
+                            :modelValue="field.value"
+                            placeholder="Choose an option"
+                            class="w-100 custom-height"
+                            style="width: 100% !important"
+                            v-bind="field.value"
+                            @update:modelValue="handleChange"
+                            @change="resetVerifiedForestClients"
+                            :class="{
+                                'is-invalid': errors.roleId,
+                            }"
+                        />
+                    </Field>
+                    <ErrorMessage class="invalid-feedback" name="roleId" />
+                </StepContainer>
+
+                <StepContainer
+                    v-if="isAbstractRoleSelected()"
+                    title="Organization information"
+                    subtitle="Associate one or more Client IDs to this user"
+                    :divider="false"
+                >
+                    <ForestClientInput
+                        :userId="formData.userId"
+                        :roleId="formData.roleId"
+                        @setVerifiedForestClients="setVerifiedForestClients"
+                        @removeVerifiedForestClients="
+                            removeVerifiedForestClients
+                        "
+                        @resetVerifiedForestClients="resetVerifiedForestClients"
+                    />
+                </StepContainer>
+
+                <div class="button-stack">
+                    <Button
+                        type="button"
+                        id="grantAccessCancel"
```

Comment:
also the id for two buttons, need to change the button id in the `GrantApplicationAdmin.vue` component as well, those are also copied

Reply:
Added unique IDs to buttons